observability/logging.go

```diff
@@ -0,0 +1,310 @@
+/*
+ *
+ * Copyright 2022 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package observability
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"unsafe"
+
+	"github.com/google/uuid"
+	binlogpb "google.golang.org/grpc/binarylog/grpc_binarylog_v1"
+	iblog "google.golang.org/grpc/internal/binarylog"
+	"google.golang.org/grpc/internal/grpcutil"
+	configpb "google.golang.org/grpc/observability/internal/config"
+	grpclogrecordpb "google.golang.org/grpc/observability/internal/logging"
+)
+
+// translateMetadata translates the metadata from Binary Logging format to
+// its GrpcLogRecord equivalent.
+func translateMetadata(m *binlogpb.Metadata) *grpclogrecordpb.GrpcLogRecord_Metadata {
+	var res grpclogrecordpb.GrpcLogRecord_Metadata
+	res.Entry = make([]*grpclogrecordpb.GrpcLogRecord_MetadataEntry, len(m.Entry))
+	for i, e := range m.Entry {
+		res.Entry[i] = &grpclogrecordpb.GrpcLogRecord_MetadataEntry{
+			Key:   e.Key,
+			Value: e.Value,
+		}
+	}
+	return &res
+}
+
+func setPeerIfPresent(binlogEntry *binlogpb.GrpcLogEntry, grpcLogRecord *grpclogrecordpb.GrpcLogRecord) {
+	if binlogEntry.GetPeer() != nil {
+		grpcLogRecord.PeerAddress = &grpclogrecordpb.GrpcLogRecord_Address{
+			Type:    grpclogrecordpb.GrpcLogRecord_Address_Type(binlogEntry.Peer.Type),
+			Address: binlogEntry.Peer.Address,
+			IpPort:  binlogEntry.Peer.IpPort,
+		}
+	}
+}
+
+var loggerTypeToEventLogger = map[binlogpb.GrpcLogEntry_Logger]grpclogrecordpb.GrpcLogRecord_EventLogger{
+	binlogpb.GrpcLogEntry_LOGGER_UNKNOWN: grpclogrecordpb.GrpcLogRecord_LOGGER_UNKNOWN,
+	binlogpb.GrpcLogEntry_LOGGER_CLIENT:  grpclogrecordpb.GrpcLogRecord_LOGGER_CLIENT,
+	binlogpb.GrpcLogEntry_LOGGER_SERVER:  grpclogrecordpb.GrpcLogRecord_LOGGER_SERVER,
+}
+
+type observabilityBinaryMethodLogger struct {
```

Comment:
s/observability// since we are in the observability package and there should be no confusion.